### PR TITLE
Add link to pod list page, allow config.js to be customized in dev mode ...

### PR DESCRIPTION
**Pull request by gashcrumb**
...via env variables, add /console/java/ mapping

Rebuild bindata

Fix typo

Update to openshift-jvm 1.0.10 for some font path fixes

### CODE REVIEW

File: assets/bower.json [hunk 1]
```diff
@@ -26,7 +26,8 @@
     "zeroclipboard": "2.2.0",
     "messenger": "1.4.1",
     "kubernetes-label-selector": "0.0.1",
-    "openshift-object-describer": "0.0.7"
+    "openshift-object-describer": "0.0.7",
+    "hawtio-utilities": "~2.0.21"
```
Comment (liggitt) [inline assets/bower.json]: lock to a specific version

File: assets/Gruntfile.js [hunk 2]
```diff
@@ -81,13 +81,45 @@ module.exports = function (grunt) {
...
           middleware: function (connect) {
             return [
-              modRewrite(['!^/(config.js|favicon.ico|(bower_components|scripts|images|styles|views)(/.*)?)$ /index.html [L]']),
+              modRewrite(['!^/(config.js|favicon.ico|(bower_components|scripts|images|styles|views|console|console/java)(/.*)?)$ /index.html [L]']),
+              connect().use('/config.js', function(req, res, next) {
+                var KUBERNETES_HOST = process.env.KUBERNETES_HOST || 'localhost';
+                var KUBERNETES_PORT = process.env.KUBERNETES_PORT || '8443';
+                var KUBERNETES_PROXY_VERSION = process.env.KUBERNETES_PROXY_VERSION || 'v1beta3';
+
+                var answer = 'window.OPENSHIFT_CONFIG = {' +
```
Comment (liggitt) [inline assets/Gruntfile.js]: this will be a pain to keep up to date with changes in config.js... not sure I like this

File: assets/Gruntfile.js [hunk 3]
```diff
@@ -81,13 +81,45 @@ module.exports = function (grunt) {
...
+                '/console/java',
+                connect.static('./openshift-jvm')
+              ),
               connect.static('.tmp'),
               connect().use(
                 '/bower_components',
                 connect.static('./bower_components')
               ),
-              connect.static(appConfig.app)
+              connect.static(appConfig.app),
```
Comment (liggitt) [inline assets/Gruntfile.js]: trailing `,`

**Comment by gashcrumb** [comment]
Example screenshot:

java-link

File: assets/app/index.html [hunk 4]
```diff
@@ -16,12 +16,59 @@
     <!-- build:css(.tmp) styles/main.css -->
     <link rel="stylesheet" type="text/css" href="styles/main.css">
     <!-- endbuild -->
-     <style type="text/css">
+    <style type="text/css">
 
 
     </style>
+    <!-- build:js(.) scripts/oldieshim.js -->
```
Comment (liggitt) [inline assets/app/index.html]: can you put the script location back where it was for now? I'd prefer to make that change in it's own commit, rather than along-side a big change... makes it easier to track down resulting problems

File: assets/app/index.html [hunk 5]
```diff
@@ -71,104 +118,60 @@
 
     <script src="config.js"></script>
```
Comment (liggitt) [inline assets/app/index.html]: config.js needs to load before the app does

File: assets/app/scripts/controllers/pods.js [hunk 6]
```diff
@@ -20,8 +20,48 @@ angular.module('openshiftConsole')
     $scope.emptyMessage = "Loading...";
     var watches = [];
 
+    $scope.gotoContainerView = function(container, jolokiaUrl) {
+      // console.log("OPENSHIFT_CONFIG: ", OPENSHIFT_CONFIG);
+      // console.log("Clicked! container: ", container, " jolokiaUrl: ", jolokiaUrl);
+      var returnTo = window.location.href;
+      var title = Core.pathGet(container, ['name']) || 'Untitled Container';
+      var targetURI = new URI();
+      targetURI.path('/console/java/')
```
Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: can't assume "/console/" prefix... that context root is configurable

File: assets/app/index.html (hunk 5 above)
Comment (gashcrumb) [inline assets/app/index.html]: To be honest if that ordering matters I'd recommend doing it as a pre-bootstrap task, for example -> https://github.com/fabric8io/fabric8-console/blob/master/plugins/main/ts/mainPlugin.ts#L124-137

File: assets/app/index.html (hunk 4 above)
Comment (gashcrumb) [inline assets/app/index.html]: Sure!

File: assets/Gruntfile.js (hunk 2 above)
Comment (spadgett) [inline assets/Gruntfile.js]: It probably belongs in a separate pull request either way.

Comment (gashcrumb) [inline assets/Gruntfile.js]: Kinda stinks though when you just want to run the console in dev mode against an instance running elsewhere.  I can take it out I guess.  Though ideally you'd maybe just read config.js off the disk and allow environment variables to overwrite defaults in config.js.  Anyway, will take it out.

File: assets/app/scripts/controllers/pods.js (hunk 6 above)
Comment (gashcrumb) [inline assets/app/scripts/controllers/pods.js]: Ok, where does that come from then?

Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: all links should be made relative to the internal root... see https://github.com/openshift/origin/tree/master/assets#navigation

File: assets/app/scripts/controllers/pods.js [hunk 7]
```diff
@@ -20,8 +20,48 @@ angular.module('openshiftConsole')
...
+                 returnTo: returnTo
+               });
+      // console.log("TargetURI: ", targetURI.toString());
+      window.location.href = targetURI.toString();
+    };
+
     watches.push(DataService.watch("pods", $scope, function(pods) {
       $scope.unfilteredPods = pods.by("metadata.name");
+      // Add a jolokia URL for any container that has a jolokia port
+      _.forIn($scope.unfilteredPods, function(pod, id) {
```
Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: generally, I don't think `Core.pathGet` makes this more readable... can you just reference the JS objects?

File: assets/app/scripts/controllers/pods.js [hunk 8]
```diff
@@ -20,8 +20,48 @@ angular.module('openshiftConsole')
...
+        var namespace = Core.pathGet(pod, ['metadata', 'namespace']) || 'default';
```
Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: are you actually seeing a missing namespace? I don't think defaulting to "default" will generally work

File: assets/app/scripts/controllers/pods.js [hunk 9]
```diff
@@ -20,8 +20,48 @@ angular.module('openshiftConsole')
...
+          if (!Core.pathGet(containerState, ['state', 'running'])) {
+            return;
+          }
+          var ports = Core.pathGet(container, ['ports']);
+          if (ports && ports.length > 0) {
+            var jolokiaPort = _.find(ports, function(port) {
+              return port.name && port.name.toLowerCase() === 'jolokia';
+            });
+            if (jolokiaPort) {
+              container.jolokiaUrl = UrlHelpers.join(ProxyPod(namespace, id, jolokiaPort.containerPort), 'jolokia') + '/';
```
Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: we're trying not to modify objects returned from the API... would prefer to see this check done closer to where the rendering is done

File: assets/app/views/_pod-template.html [hunk 10]
```diff
@@ -48,6 +48,13 @@
     </div>
 
     <div ng-if="container.ports.length > 0" class="pod-template-ports">
+      <div>
+        <span ng-show="container.jolokiaUrl" class="pull-right">
```
Comment (liggitt) [inline assets/app/views/_pod-template.html]: I thought we were going to inject the link inside the pod template itself using an extension point

File: assets/app/scripts/controllers/pods.js (hunk 6 above)
Comment (gashcrumb) [inline assets/app/scripts/controllers/pods.js]: Can't actually use $location here as we actually want to go to a different page and not a different route.  But I can get the base.href value and use that to get the right URL.

File: assets/app/views/_pod-template.html (hunk 10 above)
Comment (liggitt) [inline assets/app/views/_pod-template.html]: Seems like a good place to use https://github.com/openshift/origin/tree/master/assets#inject-additional-content-into-the-page

File: assets/app/scripts/controllers/pods.js (hunk 7 above)
Comment (gashcrumb) [inline assets/app/scripts/controllers/pods.js]: It's really handy for avoiding null references when you really want to check some attribute inside an object.  Basically avoids having a bunch of nested 'if' statements.

File: assets/app/scripts/controllers/pods.js (hunk 8 above)
Comment (gashcrumb) [inline assets/app/scripts/controllers/pods.js]: k, can take out this '||'

File: assets/app/scripts/controllers/pods.js (hunk 9 above)
Comment (gashcrumb) [inline assets/app/scripts/controllers/pods.js]: I actually would have rather done it in the model so it'd be available to other views to be honest, wouldn't it be more handy to enrich the data returned from the API for this kind of stuff?

Guess I could store it in a separate map and look it up that way...

File: assets/app/views/_pod-template.html (hunk 10 above)
Comment (gashcrumb) [inline assets/app/views/_pod-template.html]: k, guess we can do that...

File: assets/app/scripts/controllers/pods.js (hunk 9 above)
Comment (liggitt) [inline assets/app/scripts/controllers/pods.js]: If the check is well-factored, and you can run the check at the point you render the link, you don't need to worry about which controller loaded the data from the API
